Apps/Examples/Examples/All Examples/Custom2DPuckExample.swift

```diff
@@ -4,27 +4,125 @@ import MapboxMaps
 @objc(Custom2DPuckExample)
 public class Custom2DPuckExample: UIViewController, ExampleProtocol {
 
-    internal let toggleAccuracyRadiusButton: UIButton = UIButton(frame: .zero)
     internal var mapView: MapView!
-    internal var showsAccuracyRing: Bool = false {
+    internal var puckConfiguration = Puck2DConfiguration.makeDefault(showBearing: true)
+    private var showsPuck: PuckVisibility = .isVisible {
         didSet {
-            syncPuckAndButton()
+            mapView.location.options.puckType = showsPuck == .isVisible ? .puck2D(puckConfiguration) : .none
+        }
+    }
+
+    private var puckImage: PuckImage = .blueDot {
+        didSet {
+            puckConfiguration.topImage = puckImage == .blueDot ? .none : UIImage(named: "star")
+            // Only add puck with new image to the map if it is meant to be visible
+            if showsPuck == .isVisible {
+                mapView.location.options.puckType = .puck2D(puckConfiguration)
+            }
+        }
+    }
+
+    private var showsBearing: PuckBearingVisibility = .isVisible {
+        didSet {
+            // It's necessary to create a new default when switching bearing visibility
+            var newPuck = showsBearing == .isVisible ? Puck2DConfiguration.makeDefault(showBearing: true) : Puck2DConfiguration.makeDefault(showBearing: false)
```

Comment:
I would like you to rewrite this whole approach with `didSet`'s, but this particular thing if your implementation had to be the following:
```suggestion
            var newPuck = Puck2DConfiguration.makeDefault(showBearing: showsBearing == .isVisible) 
```